packages/vaadin-grid/test/array-data-provider.test.js

```diff
@@ -80,16 +80,17 @@ describe('array data provider', () => {
     it('should handle null', () => {
       grid.items = null;
       expect(grid.size).to.equal(0);
+      expect(grid.items).to.be.null;
     });
 
     it('should set array data provider', () => {
       expect(grid.dataProvider).to.equal(grid._arrayDataProvider);
     });
 
-    it('should not override custom data provider', () => {
+    it('should override custom data provider', () => {
```

Comment:
Agreed internally that setting `grid.items = people` is basically a shorthand for setting `grid.dataProvider = createArrayDataProvider(people)` so setting `items` should actually override any existing `dataProvider`